### ISSUE TITLE
Weird login failures possibly related to capitalization and/or localStorage state

### ISSUE THREAD

**Issue by callahad** [mozilla/persona#2866]
See: https://support.mozilla.org/en-US/questions/943789

When the user selects their email address an "invalid email" error shows up in the console, and nothing happens on the page.

This used to exist in `resources/static/dialog/js/misc/state.js` before the big BigTent merge. That code was:

``` javascript
handleState("email_chosen", function(msg, info) {
  var email = info.email,
      idInfo = storage.getEmail(email);

  self.email = email;

  function oncomplete() {
    complete(info.complete);
  }

  if (!idInfo) {
    throw new Error("invalid email");
  }
  // ... snip ...
}
```

One of the folks involved noted in a separate email:

> So when I try to login with "Foo@example.com" that's when I get the error.
> But when I try to login with "foo@example.com" then everything works ok!

So it sounds like we should downcase sooner? Still need to reproduce and trace where this code moved since the BigTent core merge.

**Comment by shane-tomlinson** [comment]
Link to #2551

**Comment by jamesshore** [comment]
+1. This affected at least one user of mine in production.

It's easy to reproduce:
1. Create a test user, (e.g., `http get http://personatestuser.org/email`. My test user was `lopez19464@personatestuser.org` with password `LPrAaF2ehXweXEtj`.)
2. Log in using Persona (e.g., via the sign-in link at https://developer.mozilla.org/en-US/ )
3. (Select 'This is not me' link if you're recognized by Persona.)
4. Enter test user's email address using different capitalization and press enter. (e.g., `LOPEZ19464@personatestuser.org`.)
5. Enter password and press enter. (e.g., `LPrAaF2ehXweXEtj`.)
6. Observe 'Error: invalid email' in console.

(Reproduced on Firefox 17.0.1 on Mac OS X 10.8.2, but also observed on IE9 & Chrome 24 in Windows 7, and on iPad 3.)

![Screen Shot 2013-01-02 at 11 27 49 AM](https://f.cloud.github.com/assets/123129/39069/8bbf292c-5512-11e2-9712-bb012e54c14d.png)

**Comment by callahad** [comment]
Oh, beautiful. I'm able to reproduce an error with those steps in prod, stage, and dev. Thanks, @jamesshore!

**Comment by skinny97214** [comment]
What are the issues with always downcasing?

**Comment by jrgm** [comment]
@skinny97214 - @6a68 posted to dev-identity summing up the choices. Read that (short answer: in theory the local part (username) may be case sensitive for a given email server endpoint).

**Comment by jaredhirsch** [comment]
fixed by #3078
